### PR TITLE
#43 fixing parsing for contact number to be searched in DB and on SMS

### PR DESCRIPTION
**Pull request by Landry333**
#43, #110 

 fixing parsing for contact number to be searched in DB in the searchContactToSupervise activity and on SMS autofill display in the sendSmsInvitation activity

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java [hunk 1]
```diff
@@ -88,10 +88,23 @@ private void loadContacts() {
...
+                } else {
+                    contactNumber = contactDetails
+                            .split("\\n")[1]
+                            .replaceAll("[^+0-9]", "");
+                }
+
 
                 db.collection("users")
-                        .whereEqualTo("phoneNumber", number.replaceAll("[^+0-9]", ""))
+                        //.whereEqualTo("phoneNumber", number.replaceAll("[^+0-9]", ""))
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: you can remove this comment

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java [hunk 2]
```diff
@@ -88,10 +88,23 @@ private void loadContacts() {
         listContactsView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             // argument position gives the index of item which is clicked
             public void onItemClick(AdapterView<?> arg0, View v, int position, long arg3) {
-                String number = (String) listContactsView.getItemAtPosition(position);
+                //String number = (String) listContactsView.getItemAtPosition(position);
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: I believe you can remove this comment.

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java (hunk 1 above)
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: Same thing as above; I think you can remove this comment too.

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java (hunk 2 above)
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: same remove commented code

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java [hunk 3]
```diff
@@ -82,74 +71,17 @@ public void onComplete(@NonNull Task<QuerySnapshot> task) {
                                     } else {
                                         Toast.makeText(SearchContactsByPhone.this, "User doesn't have the app", Toast.LENGTH_SHORT).show();
                                         Intent intent = new Intent(SearchContactsByPhone.this, SendSmsInvitationActivity.class);
-                                        intent.putExtra("smsNumber", smsNumber);
-                                        startActivity(intent);
-                                    }
-                                }
-                            }
-                        });
-            }
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java:91]: Thanks for removing unused code!

**Review by ShawnGuerra-Bautista — CHANGES_REQUESTED** [review]
The bug is eliminated and everything works as expected. Excellent.
There are just a few changes required.

**Review by refatK — CHANGES_REQUESTED** [review]
Good changes and nice clean up. Just requesting you remove some leftover comments and it should be good.

Also please link to the bug issue directly next time.

**Review by KennanHoa — APPROVED** [review]
Good fix. Once the redundant lines are removed (as mentioned by the 2 other reviews), it can be merged

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java (hunk 3 above)
Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java:91]: My pleasure... :)

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java (hunk 2 above)
Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: ok. It has been removed

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java (hunk 1 above)
Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: ok. It has been removed

Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: ok. It has been removed

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java (hunk 2 above)
Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsToSupervise.java]: ok. It has been removed

**Review by ShawnGuerra-Bautista — APPROVED** [review]

**Comment by refatK** [comment]
Now that we have a Code Quality checker, it pointed out some thigs you should clean up, so you should clean them up to keep quality as high as possible:
https://sonarcloud.io/project/issues?assignees=Landry333%40github&id=Landry333_Big-Owl&pullRequest=161&resolved=false&types=CODE_SMELL

**Review by refatK — APPROVED** [review]
Thanks for addressing the changes I asked for.

**Pull request merged**